lib/src/front_end/piece_writer.dart

```diff
@@ -180,14 +196,31 @@ class PieceWriter {
       _currentText = TextPiece();
     }
 
-    _write(lexeme ?? token.lexeme, offset: token.offset);
+    lexeme ??= token.lexeme;
+
+    _write(lexeme, offset: token.offset);
+  }
+
+  /// Writes multi-line [text] to the current [TextPiece].
```

Comment:
What `[text]` is this referencing? Should this be `[lexeme]`? Or IMO change the argument name back to `text`.

Reply:
Oops, yeah, copy/paste error. Changed to `text`.